### PR TITLE
build: update gh action ci

### PR DESCRIPTION
**Pull request by aladdin-add**
1. yarn => pnpm
2. upgrade gh actions actions/checkout@v2 => actions/checkout@v4
3. upgrade gh actions actions/setup-node@v2 => actions/setup-node@v3
4. test node version: 14.x, 16.x => lts/*, latest

##### Checklist

- [ ] `npm test` passes
- [ ] tests are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change

- any feature?
- close https://github.com/umijs/qiankun/ISSUE_URL

### CODE REVIEW

**Comment by kuitos** [comment]
单测可以先注释掉，可以优先修一下 monorepo 的 lint？不然新提交的 pr 风格都不统一

File: .eslintignore [hunk 1]
```diff
@@ -1,3 +1,6 @@
 examples
 dist
 writable-dom
+
+# not linting test files
```
Comment (kuitos) [inline .eslintignore:5]:
```suggestion
# TODO not linting test files temporary
```
这个应该是临时的，我加个 TODO

**Review by kuitos — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -12,11 +12,11 @@
     "build": "pnpm -r run build",
     "prerelease:alpha": "changeset pre enter alpha && changeset && changeset version",
     "release:alpha": "pnpm run build && changeset publish && changeset pre exit",
-    "lint": "eslint packages/*/src/**",
+    "lint": "eslint packages/",
```
Comment (kuitos) [inline package.json:15]: lint 里要加一下 prettier -c 目前 eslint 关闭了 code style 检查

**Review by kuitos — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (aladdin-add) [reply package.json:15]: ops, 刚刚合并了，等会重新提交一个

Comment (kuitos) [reply package.json:15]: 或者单独加一个 prettier check 的 script 然后补到 ci 里

**Pull request merged**